### DISCUSSION TITLE
Ejabberd contact list grouping and show contact list if its connected with parent entity

### DISCUSSION THREAD

**Discussion by pinanklakhani** [processone/ejabberd#4002] [Questions] [answered]
System has 2 type of Entities,

1. EntityType1
2. EntityType2

Now system has 
for ex. 

**2 EntityType1**

1. EntityType-1.1 - 5 Users   - Mark wood
2. EntityType-1.2 - 5 Users   - Mark wood

&

**2 EntityType2**

1. EntityType2.1 - 3 Users
2. EntityType2.2 - 3 Users 

Now EntityType1 can connect with EntityType2. So Once the EntityType-1.1 is connected with EntityType2.1, EntityType-1.1 users can chat with its 5 users + EntityType2.1 entitys 3 users. Now EntityType-1.1 sends connection request to EntityType2.2 and got connected. So now EntityType-1.1 can chat with EntityType-1.1 employyes, EntityType2.1 users & EntityType2.2 users.

Now, Mark wood has access of 2 employers. EntityType-1.1 & EntityType-1.2.
But it should show contact list entity wise. So When Mark is working under EntityType-1.1 it should show chat users of EntityType-1.1, EntityType2.1 & EntityType2.2.
And when mark is working for EntityType-1.2 it should show chat users of EntityType-1.2 only.

Can somebody please guide me with the reference if known

**Comment by badlop** [comment]
This is not an issue inside ejabberd, it's a question about how to configure it to suit your specific needs. So, I've moved it from Issues to Discussions.

**Comment by pinanklakhani** [comment]
@badlop Thanks for update.
Can you please guide me how can i achieve this i tried rooms, shared_roster_groups but in both cases users accessible in entity. 
In room, i created room using `create_room_with_opts ` and added users using `set_room_affiliation` but it is not showing in database. Also unable to figure out how can i integrate it with strophe.js

**Reply by pinanklakhani** [reply]
@Licaon_Kter @loe @duritong @otaviof @badlop Any inputs guys?

**Reply by badlop** [reply, marked as answer]
>  i created room using create_room_with_opts  and added users using set_room_affiliation but it is not showing in database

Only rooms configured as `persistent` are stored in the database.
